client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java

```diff
@@ -46,58 +32,45 @@
  * interface.
  */
 public final class RestProxy implements InvocationHandler {
-    private final String host;
     private final HttpClient httpClient;
     private final SerializerAdapter<?> serializer;
-    private final SwaggerInterfaceProxyDetails interfaceDetails;
+    private final SwaggerInterfaceParser interfaceParser;
 
-    private RestProxy(String host, HttpClient httpClient, SerializerAdapter<?> serializer, SwaggerInterfaceProxyDetails interfaceDetails) {
-        this.host = host;
+    private RestProxy(HttpClient httpClient, SerializerAdapter<?> serializer, SwaggerInterfaceParser interfaceParser) {
         this.httpClient = httpClient;
         this.serializer = serializer;
-        this.interfaceDetails = interfaceDetails;
+        this.interfaceParser = interfaceParser;
     }
 
     @Override
     public Object invoke(Object proxy, final Method method, Object[] args) throws Throwable {
-        final String methodName = method.getName();
-        final SwaggerMethodProxyDetails methodDetails = interfaceDetails.getMethodProxyDetails(methodName);
-
-        final String actualHost = methodDetails.applyHostSubstitutions(host, args);
-        final String[] hostParts = actualHost.split("://");
-        final String actualPath = methodDetails.getSubstitutedPath(args);
+        final SwaggerMethodParser methodParser = interfaceParser.methodParser(method);
```

Comment:
I understand your concern about performance, but this should actually be faster. If we do everything all at once, then we're probably going to be doing a lot of extra work that the user won't need. I imagine they'll only call a few methods on a given interface, in which case it would be better to only parse the ones they request. Once we parse a method, we do cache the parse so the next time it will just be a lookup.